packages/experimental/Button/SPEC.md

```diff
@@ -0,0 +1,184 @@
+   * - 'primary': Emphasizes the button as a primary action.
+   * - 'subtle': Minimzes emphasis to blend into the background until hovered or focused.
+   */
+  appearance?: 'primary' | 'outline' | 'subtle' | 'transparent';
+
+  /**
+   * Icon slot that, if specified, renders an icon either before or after the `children` as specified by the
+   * `iconPosition` prop.
+   */
+  icon?: IconSourcesType;
+
+  /**
+   * Loader slot that, if specified, renders a `loader` before the `icon` and `children` while the `loading` flag
+   * is set to `true`.
+   */
+  loader?: ActivityIndicator;
+
+  /**
+   * A button can fill the width of its container.
+   * @default false
+   */
+  block?: boolean;
+
+  /**
+   * A button can show that it cannot be interacted with.
+   * @default false
+   */
+  disabled?: boolean;
+
+  /**
+   * A button can format its icon to appear before or after its content.
+   * @default 'before'
+   */
+  iconPosition?: 'before' | 'after';
+
+  // /**
+  //  * A button can show a loading indicator if it is waiting for another action to happen before allowing itself to
+  //  * be interacted with.
+  //  * @default false
+  //  */
+  loading?: boolean;
+
+  /**
+   * A button can be rounded, circular, or square.
+   * @default 'rounded'
+   */
+  shape?: 'rounded' | 'circular' | 'square';
+
+  /**
+   * A button supports different sizes.
+   * @default 'medium'
+   */
+  size?: 'small' | 'medium' | 'large';
+};
+```
+
+### Styling Tokens
+
+TBD once we decide on tokens set.
+
+## Behaviors
+
+### States
+
+The following section describes the different states in which a `Button` can be throughout the course of interaction with it.
+
+#### Enabled and Disabled states
+
+An enabled `Button` communicates interaction by having styling that invites the user to click/tap on it to trigger an action.
+
+A disabled `Button` is non-interactive, disallowing the user to click/tap on it to trigger an action.
+
+#### Hovered state
+
+A hovered `Button` changes styling to communicate that the user has placed a cursor above it.
+
+#### Focused state
+
+A focused `Button` changes styling to communicate that the user has placed keyboard focus on it. This styling is usually the same to the one in the hovered state plus extra styling on the outline to indicate keyboard focus has been placed on the component.
+
+#### Pressed state
+
+A pressed `Button` changes styling to communicate that the user is currently pressing it.
+
+#### Loading state
+
+A loading `Button` renders a `loader` before all the other content to indicate that it is waiting for another action before allowing itself to be interacted with.
+
+### Interaction
+
+#### Keyboard interaction
+
+The following is a set of keys that interact with the `Button` component:
+
+| Key     | Description                   |
+| ------- | ----------------------------- |
+| `Enter` | Executes the `Button` action. |
```

Comment:
"Action passed into onClick" maybe, since we don't have an action prop?

Reply:
Reworded to Executes the function passed into the `onClick` prop. Thanks!